Mathlib/Algebra/Category/Grp/Ulift.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2024 Sophie Morel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sophie Morel
+-/
+import Mathlib.Algebra.Category.Grp.Limits
+import Mathlib.CategoryTheory.Limits.Preserves.Ulift
+import Mathlib.CategoryTheory.Preadditive.AdditiveFunctor
+
+/-!
+This file shows that the functors `Grp.uliftFunctor` and `CommGrp.uliftFunctor`
+(as well as the additive versions) are fully faithful, preserves all limits and
+create small limits.
+
+It also shows that `AddCommGrp.uliftFunctor` preserves zero morphisms and is an additive functor.
+
+-/
+
+universe v w w' u
+
+open CategoryTheory Limits
+
+namespace Grp
+
+/-- The universe lift functor for groups is faithful.
+-/
+@[to_additive
+  "The universe lift functor for additive groups is faithful."]
+instance : uliftFunctor.{u, v}.Faithful where
```

Comment:
The faithful and full instances can be deduced from this:
```lean
@[to_additive]
def uliftFunctorFullyFaithful : uliftFunctor.{u, v}.FullyFaithful where
  preimage f := Grp.ofHom (MulEquiv.ulift.toMonoidHom.comp
    (f.comp MulEquiv.ulift.symm.toMonoidHom))
  map_preimage _ := rfl
  preimage_map _ := rfl
```

Reply:
Done.